src/main/java/com/teamfive/disscard/DissCardController.java

```diff
@@ -58,54 +66,54 @@ public String index(Model model) {
      * Fetches a list of all cards registered with DissCard
      * @return A list of JSON objects representing cards
      */
-    @GetMapping("/card/list")
+    @GetMapping(value = "/card/list", produces = MediaType.APPLICATION_JSON_VALUE)
     @ResponseBody
     public List<Card> fetchAllCards() {
         return cardService.getAll();
     }
 
     /**
      * Fetches a card's info based on its ID
-     * @param id id of a card registered with DissCard
-     * @return A JSON object representing a card
+     * @param id ID of a card registered with DissCard
+     * @return A JSON object representing a card or 404 if not found
      */
-    @GetMapping("/card/info/{id}")
+    @GetMapping(value = "/card/info/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<Card> fetchCardById(@PathVariable("id") int id) {
         Card fetchedCard = cardService.getById(id);
-        HttpHeaders headers = new HttpHeaders();
-        headers.setContentType(MediaType.APPLICATION_JSON);
-        return new ResponseEntity<>(fetchedCard, headers, HttpStatus.OK);
+        if (fetchedCard == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(fetchedCard, HttpStatus.OK);
     }
 
     /**
      * Searches DissCard's systems for a list of cards that match the specified keyword
      * @param keyword String used to filter cards
-     * @return A list of card JSON objects that match the specified keyword
+     * @return A list of card JSON objects that match the specified keyword or 404 if none found
      */
-    @GetMapping("/card/search/{keyword}")
+    @GetMapping(value = "/card/search/{keyword}", produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity<List<Card>> fetchCardsByKeyword(@PathVariable("keyword") String keyword) {
         List<Card> fetchedCards = cardService.searchByName(keyword);
-        HttpHeaders headers = new HttpHeaders();
-        headers.setContentType(MediaType.APPLICATION_JSON);
-        return new ResponseEntity<>(fetchedCards, headers, HttpStatus.OK);
+        if (fetchedCards == null || fetchedCards.isEmpty()) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(fetchedCards, HttpStatus.OK);
```

Comment:
Added null check to handle case where card is not found and empty check for search results:
Source: https://www.tutorialspoint.com/java/java_class_attributes.htm